### PR TITLE
Retrieve new respondent conversations

### PR DESCRIPTION
**Pull request by AndrewTorrance**
**What is the context of this PR?**
The requirement is that internal users can see new conversations from respondents on a new tab.
Once replied to they disappear from the new tab, so that new conversation requests do not go unnoticed

Link to Trello card
https://trello.com/c/G1PNU5Qa

Describe what you have changed and why, link to other PRs or Issues as appropriate.
Both the get threads and get threads count end points needed to support a new parameter for those conversations where a respondent started a new conversation

**How to review**
Run the new feature tests . These exercise the new functionality

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -443,7 +443,7 @@ Note, the response will have either uuids or 'GROUP' for the user ids, and a uui
 For descriptions of @msg_from, @msg_to and @ru see messages get
 [Get Message List](#get-message-list)
 
-## Get Conversations count  
+h2: Get Conversations count
```
Comment (insacuri) [inline API.md]: I know this is super picky, but we use ## for a second level heading in every other repo, could we keep it consistent?

Comment (AndrewTorrance) [reply API.md]: sure

File: API.md [hunk 2]
```diff
@@ -443,7 +443,7 @@ Note, the response will have either uuids or 'GROUP' for the user ids, and a uui
 For descriptions of @msg_from, @msg_to and @ru see messages get
 [Get Message List](#get-message-list)
 
-## Get Conversations count  
+h2: ## Get Conversations count
```
Comment (insacuri) [inline API.md]: Also need to remove the h2 ^^

**Comment by insacuri** [comment]
I think having 'my_conversations' and 'new_respondent_conversations' as boolean parameters is wrong.  I can't explain why (and I'm just doing some research on it now) but I think theres a cleaner way to do this.
One of my initial ideas could be to have it instead of being an explicit flag that you pass, you can instead have a length (though maybe that's wrong...) param so you can do`/threads?length=X`.
When combined with another param (`?started_by=internal/external` as a rough idea) you can get all the threads that have 1 message in them, and were started by an external user.  That would get you the same detail and be usable by other calls in the future.

File: API.md (hunk 2 above)
Comment (AndrewTorrance) [reply API.md]: It isnt there ...... The md viewer is adding it

**Comment by insacuri** [comment]
Code looks good, just need to test it.

**Pull request merged**